osquery/tables/system/darwin/smc_keys.cpp

```diff
@@ -315,5 +316,148 @@ QueryData genSMCKeys(QueryContext &context) {
+    {"TC5C", "CPU Core 5"},
+    {"TC6C", "CPU Core 6"},
+    {"TC7C", "CPU Core 7"},
+    {"TC8C", "CPU Core 8"},
+    {"TCAH", "CPU 1 Heatsink Alt."},
+    {"TCAD", "CPU 1 Package Alt."},
+    {"TC1P", "CPU 2 Proximity"},
+    {"TC1H", "CPU 2 Heatsink"},
+    {"TC1D", "CPU 2 Package"},
+    {"TC1E", "CPU 2"},
+    {"TC1F", "CPU 2"},
+    {"TCBH", "CPU 2 Heatsink Alt."},
+    {"TCBD", "CPU 2 Package Alt."},
+    {"TCSC", "PECI SA"},
+    {"TCSc", "PECI SA"},
+    {"TCSA", "PECI SA"},
+    {"TCGC", "PECI GPU"},
+    {"TCGc", "PECI GPU"},
+    {"TG0P", "GPU Proximity"},
+    {"TG0D", "GPU Die"},
+    {"TG1D", "GPU Die"},
+    {"TG0H", "GPU Heatsink"},
+    {"TG1H", "GPU Heatsink"},
+    {"Ts0S", "Memory Proximity"},
+    {"TM0P", "Mem Bank A1"},
+    {"TM1P", "Mem Bank A2"},
+    {"TM8P", "Mem Bank B1"},
+    {"TM9P", "Mem Bank B2"},
+    {"TM0S", "Mem Module A1"},
+    {"TM1S", "Mem Module A2"},
+    {"TM8S", "Mem Module B1"},
+    {"TM9S", "Mem Module B2"},
+    {"TN0D", "Northbridge Die"},
+    {"TN0P", "Northbridge Proximity 1"},
+    {"TN1P", "Northbridge Proximity 2"},
+    {"TN0C", "MCH Die"},
+    {"TN0H", "MCH Heatsink"},
+    {"TP0D", "PCH Die"},
+    {"TPCD", "PCH Die"},
+    {"TP0P", "PCH Proximity"},
+    {"TA0P", "Airflow 1"},
+    {"TA1P", "Airflow 2"},
+    {"Th0H", "Heatpipe 1"},
+    {"Th1H", "Heatpipe 2"},
+    {"Th2H", "Heatpipe 3"},
+    {"Tm0P", "Mainboard Proximity"},
+    {"Tp0P", "Powerboard Proximity"},
+    {"Ts0P", "Palm Rest"},
+    {"Tb0P", "BLC Proximity"},
+    {"TL0P", "LCD Proximity"},
+    {"TW0P", "Airport Proximity"},
+    {"TH0P", "HDD Bay 1"},
+    {"TH1P", "HDD Bay 2"},
+    {"TH2P", "HDD Bay 3"},
+    {"TH3P", "HDD Bay 4"},
+    {"TO0P", "Optical Drive"},
+    {"TB0T", "Battery TS_MAX"},
+    {"TB1T", "Battery 1"},
+    {"TB2T", "Battery 2"},
+    {"TB3T", "Battery"},
+    {"Tp0P", "Power Supply 1"},
+    {"Tp0C", "Power Supply 1 Alt."},
+    {"Tp1P", "Power Supply 2"},
+    {"Tp1C", "Power Supply 2 Alt."},
+    {"Tp2P", "Power Supply 3"},
+    {"Tp3P", "Power Supply 4"},
+    {"Tp4P", "Power Supply 5"},
+    {"Tp5P", "Power Supply 6"},
+    {"TS0C", "Expansion Slots"},
+    {"TA0S", "PCI Slot 1 Pos 1"},
+    {"TA1S", "PCI Slot 1 Pos 2"},
+    {"TA2S", "PCI Slot 2 Pos 1"},
+    {"TA3S", "PCI Slot 2 Pos 2"}
+  };
+
+  for (const auto& iter : keyToName) {
+    genSMCKey(iter.first, smc, results);
+
+    if (results.empty()) {
+      continue;
+    }
+
+    auto& smcRow = results.back();
+    if (smcRow["value"].empty() /*|| smcRow["type"].compare("sp78") != 0*/) {
+      results.pop_back();
+      continue;
+    }
+
+    Row r;
+    r["smc_key"] = smcRow["key"];
+    r["name"] = keyToName[smcRow["key"]];
+
+    // Convert hex string to decimal.
+    std::string val;
+    try {
+      val = boost::algorithm::unhex(smcRow["value"]);
+    } catch (const boost::algorithm::hex_decode_error& e) {
+      results.pop_back();
+      continue;
+    }
+
+    int intValue = (val[0] * 256 + val[1]) >> 2;
```

Comment:
This will be an exception if `val.size()` < 2. Add a check for size before this line to make sure you have 2 characters. Unless `unhex`'s docs guarantee at least one encoded byte?